src/insane/mcmc_cfbd.jl

```diff
@@ -14,69 +14,777 @@ Created 07 10 2021
+
+    x1 = run_newton(μ0, σ0, m, sd)
+
+    μ_rdist = (x1[1], x1[2])
+
+    # marginal likelihood
+    pp = ref_posterior(Ξ, idf, λc, μc, v, mc, th, stem, λ_prior, μ_prior, 
+                       λ_rdist, μ_rdist, nitpp, nthpp, βs, pup)
+
+    # process with reference distribution the posterior
+    p1 = Vector{Float64}(undef, size(r,1))
+    for i in Base.OneTo(size(r,1))
+      p1[i] = r[i,2] + r[i,3] - 
+              logdgamma(r[i,4], λ_rdist[1], λ_rdist[2]) -
+              logdtnorm(r[i,5], μ_rdist[1], μ_rdist[2])
+    end
+    pp[1] = p1
+
+    reverse!(pp)
+    reverse!(βs)
+
+    ml = gss(pp, βs)=#
+  else
+    ml = NaN
+  end
+
+  return r, treev, ml
 end
 
 
 
 
+"""
+    mcmc_burn_cfbd(Ξ        ::Vector{sTfbd},
+                   idf      ::Array{iBfffs,1},
+                   λ_prior  ::NTuple{2,Float64},
+                   μ_prior  ::NTuple{2,Float64},
+                   λmμ_prior::NTuple{2,Float64},
+                   ψ_prior  ::NTuple{2,Float64},
+                   nburn    ::Int64,
+                   λc       ::Float64,
+                   μc       ::Float64,
+                   ψc       ::Float64,
+                   mc     ::Float64,
+                   th     ::Float64,
+                   stem   ::Bool,
+                   pup      ::Array{Int64,1}, 
+                   prints   ::Int64)
+
+Adaptive MCMC phase for da chain for constant fossilized birth-death using 
+forward simulation.
+"""
+function mcmc_burn_cfbd(Ξ        ::Vector{sTfbd},
+                        idf      ::Array{iBfffs,1},
+                        λ_prior  ::NTuple{2,Float64},
+                        μ_prior  ::NTuple{2,Float64},
+                        λmμ_prior::NTuple{2,Float64},
+                        ψ_prior  ::NTuple{2,Float64},
+                        nburn    ::Int64,
+                        λc       ::Float64,
+                        μc       ::Float64,
+                        ψc       ::Float64,
+                        mc       ::Float64,
+                        th       ::Float64,
+                        stem     ::Bool,
+                        pup      ::Array{Int64,1}, 
+                        prints   ::Int64)
+
+  el = lastindex(idf)                  # number of branches
+  L  = treelength(Ξ)                   # tree length
+  nfos = Float64(nfossils(Ξ))          # number of fossilization events
+  ns = Float64(nnodesbifurcation(Ξ))   # number of speciation events
+  ne = 0.0                             # number of extinction events
+
+  # add simulated subtrees to all fossil tips
+  #=for bi in filter(x -> it(x) && ifos(x), idf)
+    dri = dr(bi)
+    ldr = lastindex(dri)
+    t0 = sTfbd()
+    ret = false
+    while !ret
+      t0, ret = fsψtip(bi, λc, μc, ψc)
+    end
+    swapfossil!(tree, t0, dri, ldr, 0)
+  end=#
+
+  # likelihood
+  llc = llik_cfbd(Ξ, λc, μc, ψc) + log(mc) + prob_ρ(idf)
+  prc = logdgamma(λc, λ_prior[1], λ_prior[2]) + 
+        logdgamma(μc, μ_prior[1], μ_prior[2])
+
+  if isnan(λmμ_prior[1])
```

Comment:
was this for the net diversification update? I guess with the new Gibss sampling we can remove this.

Reply:
I removed λmμ_prior everywhere